mesh/amqp.c

```diff
@@ -460,125 +593,143 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
 void mesh_amqp_free(struct mesh_amqp *amqp)
 {
 	if (amqp->thread_started) {
-		mesh_amqp_close(amqp);
+		mesh_amqp_stop(amqp);
 		pthread_join(amqp->thread, NULL);
 	}
 
 	l_io_destroy(amqp->io);
 	l_queue_destroy(amqp->queue, l_free);
-	l_free(amqp->url);
-	l_free(amqp->exchange);
-	l_free(amqp->routing_key);
+	l_queue_destroy(amqp->ret_queue, l_free);
 	l_free(amqp);
 }
 
-const char *mesh_amqp_get_url(struct mesh_amqp *amqp)
+static void *get_ret_message(struct mesh_amqp *amqp)
 {
-	return amqp->url;
+	// ELL DBus API requires that the getters functions will be blocking.
+	// To fullfill this requirements there is needed construction that
+	// allows to wait for incoming value and iterate the main loop.
```

Comment:
Wording: "ELL DBus API requires that getters are blocking, so keep iterating the loop while we wait for reply from the thread".

Also, don't use C++ style comments.

Another question: what if the thread never replies?

Reply:
Everything will live, but every call to the not responding thread will cause that the call stack will grow up. IMO some timeout can be added here.

Reply:
I've added timeout